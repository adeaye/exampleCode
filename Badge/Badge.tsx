import * as React from 'react';
import { cva, type VariantProps } from 'class-variance-authority';
import Icon, { IconName } from '@components/v1/platform/molecules/Icon';
import classNames from 'classnames';
import { cn } from '@/lib/utils';

const badgeVariants = cva(
  'inline-flex items-center gap-x-1 rounded-md border px-[6px] py-[2px] text-xs',
  {
    variants: {
      variant: {
        success: 'border-rp-green-200 bg-rp-green-50 text-rp-green-600',
        warning: 'border-rp-orange-200 bg-rp-orange-50 text-rp-orange-600',
        danger: 'border-rp-red-200 bg-rp-red-50 text-rp-red-600',
        info: 'border-rp-blue-200 bg-rp-blue-50 text-rp-blue-600',
      },
    },
    defaultVariants: {
      variant: 'success',
    },
  }
);

export interface BadgeProps
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof badgeVariants> {
  icon?: IconName;
}

function Badge({ className, variant, children, icon, ...props }: BadgeProps) {
  return (
    <div className={cn(badgeVariants({ variant }), className)} {...props}>
      {icon ? <Icon icon={icon} className={classNames('h-4 w-4')} /> : null}
      {children}
    </div>
  );
}

export { Badge, badgeVariants };
