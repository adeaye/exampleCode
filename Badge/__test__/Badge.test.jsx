import React from 'react';
import { render } from '@testing-library/react';
import { Badge } from '../Badge';

describe('Badge Component test with snapshot', () => {
  it('renders with default variant', () => {
    const { container } = render(<Badge>Default Badge</Badge>);
    expect(container).toMatchSnapshot();
  });

  it('renders with a warning variant', () => {
    const { container } = render(
      <Badge variant="warnings">Warning Badge</Badge>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with a success variant', () => {
    const { container } = render(
      <Badge variant="success">success Badge</Badge>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with a danger variant', () => {
    const { container } = render(<Badge variant="danger">danger Badge</Badge>);
    expect(container).toMatchSnapshot();
  });

  it('renders with a info variant', () => {
    const { container } = render(<Badge variant="info">info Badge</Badge>);
    expect(container).toMatchSnapshot();
  });

  it('renders with an icon', () => {
    const { container } = render(<Badge icon="Tb360">Badge with Icon</Badge>);
    expect(container).toMatchSnapshot();
  });

  it('applies custom className', () => {
    const { container } = render(
      <Badge className="custom-class">Custom Class Badge</Badge>
    );
    expect(container).toMatchSnapshot();
  });
});
