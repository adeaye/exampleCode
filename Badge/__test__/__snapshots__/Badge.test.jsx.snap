// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component test with snapshot applies custom className 1`] = `
<div>
  <div
    class="inline-flex items-center gap-x-1 rounded-md border px-[6px] py-[2px] text-xs border-rp-green-200 bg-rp-green-50 text-rp-green-600 custom-class"
  >
    Custom Class Badge
  </div>
</div>
`;

exports[`Badge Component test with snapshot renders with a danger variant 1`] = `
<div>
  <div
    class="inline-flex items-center gap-x-1 rounded-md border px-[6px] py-[2px] text-xs border-rp-red-200 bg-rp-red-50 text-rp-red-600"
  >
    danger Badge
  </div>
</div>
`;

exports[`Badge Component test with snapshot renders with a info variant 1`] = `
<div>
  <div
    class="inline-flex items-center gap-x-1 rounded-md border px-[6px] py-[2px] text-xs border-rp-blue-200 bg-rp-blue-50 text-rp-blue-600"
  >
    info Badge
  </div>
</div>
`;

exports[`Badge Component test with snapshot renders with a success variant 1`] = `
<div>
  <div
    class="inline-flex items-center gap-x-1 rounded-md border px-[6px] py-[2px] text-xs border-rp-green-200 bg-rp-green-50 text-rp-green-600"
  >
    success Badge
  </div>
</div>
`;

exports[`Badge Component test with snapshot renders with a warning variant 1`] = `
<div>
  <div
    class="inline-flex items-center gap-x-1 rounded-md border px-[6px] py-[2px] text-xs border-rp-orange-200 bg-rp-orange-50 text-rp-orange-600"
  >
    Warning Badge
  </div>
</div>
`;

exports[`Badge Component test with snapshot renders with an icon 1`] = `
<div>
  <div
    class="inline-flex items-center gap-x-1 rounded-md border px-[6px] py-[2px] text-xs border-rp-green-200 bg-rp-green-50 text-rp-green-600"
  >
    <svg
      class="h-4 w-4"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <path
        d="M17 15.328c2.414 -.718 4 -1.94 4 -3.328c0 -2.21 -4.03 -4 -9 -4s-9 1.79 -9 4s4.03 4 9 4"
      />
      <path
        d="M9 13l3 3l-3 3"
      />
    </svg>
    Badge with Icon
  </div>
</div>
`;

exports[`Badge Component test with snapshot renders with default variant 1`] = `
<div>
  <div
    class="inline-flex items-center gap-x-1 rounded-md border px-[6px] py-[2px] text-xs border-rp-green-200 bg-rp-green-50 text-rp-green-600"
  >
    Default Badge
  </div>
</div>
`;
